core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckService.java

```diff
@@ -178,15 +179,13 @@ public static HealthCheckServiceBuilder builder() {
             this.longPollingTimeoutJitterRate = longPollingTimeoutJitterRate;
             this.pingIntervalMillis = pingIntervalMillis;
             healthCheckerListener = this::onHealthCheckerUpdate;
-            pendingHealthyResponses = new ObjectLinkedOpenHashSet<>();
-            pendingUnhealthyResponses = new ObjectLinkedOpenHashSet<>();
+            pendingResponses = ConcurrentHashMap.newKeySet();
```

Comment:
nit; `pendingResponses` is always updated inside a lock already

Reply:
You're right. It seems we can still use `ObjectLinkedOpenHashSet` as the lock is already used.